File: Ent_UsAdmin/impuestos.vb
Imports System.Data.OleDb
Imports System.Data.SqlClient

Public Class impuestos
    Dim strimpuestos As String
    Private Sub btn_agregar_Click(sender As Object, e As EventArgs) Handles btn_agregar.Click
        Agregar_Impuestos.Show()

    End Sub




    Public Sub cargarimpuestos()
        dtimpuestos.Rows.Clear()
        Try

            iniciarconexionempresa()

            strimpuestos = "select * from
(select id,concat(nombre, ' ',apellidopaterno,' ',apellidomaterno) as nombre from clientes)cli where cli.nombre like '%" & txtnombre.Text & "%' order by cli.nombre asc "

            Dim ejec = New SqlCommand(strimpuestos)
            ejec.Connection = conexionempresa
            Dim id, nombre, valor, factor, tipo As String

            Dim myreaderimpuestos As SqlDataReader = ejec.ExecuteReader()
            While myreaderimpuestos.Read
                id = myreaderimpuestos("id")
                nombre = myreaderimpuestos("nombre")

                dtimpuestos.Rows.Add(id, nombre)
            End While
            myreaderimpuestos.Close()
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try

    End Sub

    Private Sub impuestos_Load(sender As Object, e As EventArgs) Handles Me.Load
        cargarimpuestos()
        txtnombre.Select()

    End Sub

    Private Sub BunifuThinButton21_Click(sender As Object, e As EventArgs) Handles BunifuThinButton21.Click
        cargarimpuestos()
    End Sub

    Private Sub VerInformaciónToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles VerInformaciónToolStripMenuItem.Click
        InformacionCliente.idCLiente = dtimpuestos.Rows(dtimpuestos.CurrentRow.Index).Cells(0).Value

        InformacionCliente.Show()
    End Sub

    Private Sub ContextMenuStrip1_Opening(sender As Object, e As System.ComponentModel.CancelEventArgs) Handles ContextMenuStrip1.Opening

    End Sub

    Private Sub dtimpuestos_CellContentClick(sender As Object, e As DataGridViewCellEventArgs) Handles dtimpuestos.CellContentClick

    End Sub

    Private Sub dtimpuestos_SelectionChanged(sender As Object, e As EventArgs) Handles dtimpuestos.SelectionChanged
        If dtimpuestos.Rows(dtimpuestos.CurrentRow.Index).Cells(1).Value <> "" Then
            dtimpuestos.ContextMenuStrip = ContextMenuStrip1

        End If
    End Sub

    Private Sub MonoFlat_Label1_Click(sender As Object, e As EventArgs) Handles MonoFlat_Label1.Click

    End Sub

    Private Sub txtnombre_OnValueChanged(sender As Object, e As EventArgs) Handles txtnombre.OnValueChanged

    End Sub

    Private Sub txtnombre_KeyDown(sender As Object, e As KeyEventArgs) Handles txtnombre.KeyDown
        If e.KeyCode = Keys.Enter Then
            cargarimpuestos()
        End If
    End Sub

    Private Sub ImprimirTarjetaToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ImprimirTarjetaToolStripMenuItem.Click
        ImprimirTarjeta.idCredito = dtimpuestos.Rows(dtimpuestos.CurrentRow.Index).Cells(0).Value

        ImprimirTarjeta.Show()
    End Sub
End Class

File: Ent_UsAdmin/Catalogos/impuestos.cs
using System;
using System.Data.SqlClient;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{

    public partial class impuestos
    {
        private string strimpuestos;

        public impuestos()
        {
            InitializeComponent();
        }
        private void btn_agregar_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.Agregar_Impuestos.Show();

        }




        public void cargarimpuestos()
        {
            dtimpuestos.Rows.Clear();
            try
            {

                Module1.iniciarconexionempresa();

                strimpuestos = @"select * from
(select id,concat(nombre, ' ',apellidopaterno,' ',apellidomaterno) as nombre from clientes)cli where cli.nombre like '%" + txtnombre.Text + "%' order by cli.nombre asc ";

                var ejec = new SqlCommand(strimpuestos);
                ejec.Connection = Module1.conexionempresa;
                string id, nombre, valor, factor, tipo;

                var myreaderimpuestos = ejec.ExecuteReader();
                while (myreaderimpuestos.Read())
                {
                    id = Conversions.ToString(myreaderimpuestos["id"]);
                    nombre = Conversions.ToString(myreaderimpuestos["nombre"]);

                    dtimpuestos.Rows.Add(id, nombre);
                }
                myreaderimpuestos.Close();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }

        }

        private void impuestos_Load(object sender, EventArgs e)
        {
            cargarimpuestos();
            txtnombre.Select();

        }

        private void BunifuThinButton21_Click(object sender, EventArgs e)
        {
            cargarimpuestos();
        }

        private void VerInformaciónToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.InformacionCliente.idCLiente = Conversions.ToInteger(dtimpuestos.Rows[dtimpuestos.CurrentRow.Index].Cells[0].Value);

            My.MyProject.Forms.InformacionCliente.Show();
        }

        private void ContextMenuStrip1_Opening(object sender, System.ComponentModel.CancelEventArgs e)
        {

        }

        private void dtimpuestos_CellContentClick(object sender, DataGridViewCellEventArgs e)
        {

        }

        private void dtimpuestos_SelectionChanged(object sender, EventArgs e)
        {
            if (Conversions.ToBoolean(Operators.ConditionalCompareObjectNotEqual(dtimpuestos.Rows[dtimpuestos.CurrentRow.Index].Cells[1].Value, "", false)))
            {
                dtimpuestos.ContextMenuStrip = ContextMenuStrip1;

            }
        }

        private void MonoFlat_Label1_Click(object sender, EventArgs e)
        {

        }

        private void txtnombre_OnValueChanged(object sender, EventArgs e)
        {

        }

        private void txtnombre_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                cargarimpuestos();
            }
        }

        private void ImprimirTarjetaToolStripMenuItem_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.ImprimirTarjeta.idCredito = Conversions.ToString(dtimpuestos.Rows[dtimpuestos.CurrentRow.Index].Cells[0].Value);

            My.MyProject.Forms.ImprimirTarjeta.Show();
        }

        private void ToolStripMenuItem1_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.ActualizarCliente.idCliente = Conversions.ToString(dtimpuestos.Rows[dtimpuestos.CurrentRow.Index].Cells[0].Value);

            My.MyProject.Forms.ActualizarCliente.Show();
        }
    }
}